lib/moderation/cancel-user-engagements.ts

```
@@ -0,0 +1,498 @@
+    work.push(
+      ...consultations.map((c) => ({ kind: "consultation" as const, id: c.id })),
+      ...subscriptions.map((s) => ({ kind: "subscription" as const, id: s.id })),
+    );
+    const webinarIds = new Set<string>();
+    const classIds = new Set<string>();
+    for (const slot of attendedSlots) {
+      if (slot.appointment?.webinarId) webinarIds.add(slot.appointment.webinarId);
+      if (slot.appointment?.classId) classIds.add(slot.appointment.classId);
+    }
+    work.push(
+      ...[...webinarIds].map((id) => ({ kind: "webinar-attendance" as const, id })),
```

ESLint

[error] Use Array.from(webinarIds, (id) => ({ kind: "webinar-attendance" as const, id })) instead of [...webinarIds].map((id) => ({ kind: "webinar-attendance" as const, id })) to avoid creating an intermediate array

(e18e/prefer-array-from-map)